### PR TITLE
Mangata dev v0.9.29

### PR DESCRIPTION
**Pull request by devdanco**

### CODE REVIEW

File: asset-registry/src/lib.rs [hunk 1]
```diff
@@ -104,25 +104,29 @@ pub mod module {
...
 		fn build(&self) {
-			for (asset_id, metadata_encoded) in self.pre_register_assets.iter() {
+			self.assets.iter().for_each(|(asset_id, metadata_encoded)| {
 				let metadata = AssetMetadata::decode(&mut &metadata_encoded[..]).expect("Error decoding AssetMetadata");
-				Pallet::<T>::do_register_asset(metadata, Some(*asset_id)).expect("Error registering Asset");
-			}
+				Pallet::<T>::do_register_asset(metadata, Some(asset_id.clone())).expect("Error registering Asset");
+			});
+
+			// LastAssetId::<T>::set(self.last_asset_id.clone());
```
Comment (iStrike7) [inline asset-registry/src/lib.rs:129]: Please remove all comment references to LastAssetId, including the storage item definition on line 103

File: xtokens/src/lib.rs [hunk 2]
```diff
@@ -199,7 +200,7 @@ pub mod module {
 		/// received. Receiving depends on if the XCM message could be delivered
 		/// by the network, and if the receiving chain would handle
 		/// messages correctly.
-		#[pallet::weight(Pallet::<T>::fixed_weight_of_transfer())]
+		#[pallet::weight(Pallet::<T>::weight_of_transfer(currency_id.clone(), *amount, dest))]
```
Comment (iStrike7) [inline xtokens/src/lib.rs:203]: Please revert this change! This was the reason xcm transfers was crashing the whole network!

File: xtokens/src/lib.rs [hunk 3]
```diff
@@ -226,7 +227,6 @@ pub mod module {
 		/// by the network, and if the receiving chain would handle
 		/// messages correctly.
 		#[pallet::weight(Pallet::<T>::weight_of_transfer_multiasset(asset, dest))]
-		#[transactional]
```
Comment (iStrike7) [inline xtokens/src/lib.rs:229]: Why is this extrinsic no longer transactional?

File: xtokens/src/lib.rs [hunk 4]
```diff
@@ -260,7 +260,7 @@ pub mod module {
 		/// received. Receiving depends on if the XCM message could be delivered
 		/// by the network, and if the receiving chain would handle
 		/// messages correctly.
-		#[pallet::weight(Pallet::<T>::fixed_weight_of_transfer())]
+		#[pallet::weight(Pallet::<T>::weight_of_transfer(currency_id.clone(), *amount, dest))]
```
Comment (iStrike7) [inline xtokens/src/lib.rs:263]: Please revert this change! This was the reason xcm transfers was crashing the whole network!

File: xtokens/src/lib.rs [hunk 5]
```diff
@@ -329,7 +328,7 @@ pub mod module {
 		/// received. Receiving depends on if the XCM message could be delivered
 		/// by the network, and if the receiving chain would handle
 		/// messages correctly.
-		#[pallet::weight(Pallet::<T>::fixed_weight_of_transfer())]
+		#[pallet::weight(Pallet::<T>::weight_of_transfer_multicurrencies(currencies, fee_item, dest))]
```
Comment (iStrike7) [inline xtokens/src/lib.rs:331]: Please revert this change! This was the reason xcm transfers was crashing the whole network!

File: xtokens/src/lib.rs [hunk 6]
```diff
@@ -298,7 +298,6 @@ pub mod module {
 		/// by the network, and if the receiving chain would handle
 		/// messages correctly.
 		#[pallet::weight(Pallet::<T>::weight_of_transfer_multiasset(asset, dest))]
-		#[transactional]
```
Comment (iStrike7) [inline xtokens/src/lib.rs:301]: Why is this extrinsic no longer transactional?

File: xtokens/src/lib.rs [hunk 7]
```diff
@@ -360,7 +359,6 @@ pub mod module {
 		/// by the network, and if the receiving chain would handle
 		/// messages correctly.
 		#[pallet::weight(Pallet::<T>::weight_of_transfer_multiassets(assets, fee_item, dest))]
-		#[transactional]
```
Comment (iStrike7) [inline xtokens/src/lib.rs:363]: Why is this extrinsic no longer transactional?

File: xtokens/src/lib.rs [hunk 8]
```diff
@@ -820,10 +818,6 @@ pub mod module {
 
 	// weights
 	impl<T: Config> Pallet<T> {
-		fn fixed_weight_of_transfer() -> Weight {
-			(2_000_000_000 as Weight).saturating_add(T::BaseXcmWeight::get())
-		}
-
```
Comment (iStrike7) [inline xtokens/src/lib.rs:826]: Please revert this change! This was the reason xcm transfers was crashing the whole network!

**Review by iStrike7 — CHANGES_REQUESTED** [review]

File: asset-registry/src/lib.rs (hunk 1 above)
Comment (devdanco) [reply asset-registry/src/lib.rs]: done

File: xtokens/src/lib.rs (hunk 2 above)
Comment (devdanco) [reply xtokens/src/lib.rs]: done

File: xtokens/src/lib.rs (hunk 3 above)
Comment (devdanco) [reply xtokens/src/lib.rs]: done

File: xtokens/src/lib.rs (hunk 4 above)
Comment (devdanco) [reply xtokens/src/lib.rs]: done

File: xtokens/src/lib.rs (hunk 5 above)
Comment (devdanco) [reply xtokens/src/lib.rs]: done

File: xtokens/src/lib.rs (hunk 6 above)
Comment (devdanco) [reply xtokens/src/lib.rs]: done

File: xtokens/src/lib.rs (hunk 7 above)
Comment (devdanco) [reply xtokens/src/lib.rs]: done

File: xtokens/src/lib.rs (hunk 8 above)
Comment (devdanco) [reply xtokens/src/lib.rs]: done

**Review by mateuszaaa — APPROVED** [review]
lgtm, just make sure that try-runtime works

**Comment by devdanco** [comment]
@mateuszaaa try-runtime works :)

**Review by iStrike7 — APPROVED** [review]

**Pull request merged**